etcdserver/api/v3rpc/lease.go

```diff
@@ -73,6 +73,17 @@ func (ls *LeaseServer) LeaseKeepAlive(stream pb.Lease_LeaseKeepAliveServer) erro
 		resp := &pb.LeaseKeepAliveResponse{ID: req.ID, Header: &pb.ResponseHeader{}}
 		ls.hdr.fill(resp.Header)
 
+		if req.Probe {
+			lresp, lerr := ls.le.LeaseProbe(lease.LeaseID(req.ID))
+			if lerr != nil {
+				return lerr
+			}
+			resp.TTL = lresp.TTL
+			resp.Remaining = lresp.Remaining
+			resp.Keys = lresp.Keys
+			return stream.Send(resp)
```

Comment:
don't bail out of the stream, just keep accepting requests like renew? otherwise this seems like it would leak